### PR TITLE
update py-ansible, py-ansible-core, py-pynacl, py-resolvelib, py-bcrypt, py-paramiko, py-httplib2

### PR DESCRIPTION
**Pull request by tsabirgaliev**
#### Description

Update ansible and dependencies

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.1 22C65 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-ansible-core/Portfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ long_description \
     be written in any language and are transferred to managed machines \
     automatically.
 
-python.versions     37 38 39 310
```
Comment (reneeotten) [inline python/py-ansible-core/Portfile:34]: did you verify that no other ports depend anymore on the py37/py38 subports? If that's really the case you should remove the select files for these two versions as well.

File: python/py-ansible/Portfile [hunk 2]
```diff
@@ -30,7 +30,7 @@ long_description \
     be written in any language and are transferred to managed machines \
     automatically.
 
-python.versions     37 38 39 310
```
Comment (reneeotten) [inline python/py-ansible/Portfile:33]: same as above - no dependents anymore?

File: python/py-bcrypt/Portfile [hunk 3]
```diff
@@ -6,11 +6,11 @@ PortGroup           select 1.0
 
 name                py-bcrypt
 version             4.0.1
-revision            0
+revision            1
```
Comment (reneeotten) [inline python/py-bcrypt/Portfile:9]: no need to increase the "revision" as there are no changes to existing subports. The new addition of py311 will be build regardless.

File: python/py-httplib2/Portfile [hunk 4]
```diff
@@ -4,8 +4,8 @@ PortSystem        1.0
 PortGroup         python 1.0
 
 name              py-httplib2
-set realversion   0.20.4
-revision          1
+set realversion   0.21.0
```
Comment (reneeotten) [inline python/py-httplib2/Portfile:7]: does this new version still support everything down to Python 2.7?

File: python/py-pynacl/Portfile [hunk 5]
```diff
@@ -7,12 +7,12 @@ PortGroup           select 1.0
 name                py-pynacl
 python.rootname     PyNaCl
 version             1.5.0
-revision            0
+revision            1
```
Comment (reneeotten) [inline python/py-pynacl/Portfile:10]: no need to increase the revision - see above

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-httplib2/Portfile (hunk 4 above)
Comment (steenzout) [reply python/py-httplib2/Portfile:7]: it seems like it https://github.com/httplib2/httplib2/blob/master/setup.py

Comment (reneeotten) [reply python/py-httplib2/Portfile:7]: great - to be honest I didn't check, was just reviewing the PR on my phone which make going to upstream websites a bit cumbersome ;)

File: python/py-ansible-core/Portfile (hunk 1 above)
Comment (tsabirgaliev) [reply python/py-ansible-core/Portfile:34]: I did a quick grep:
```
❯ find . -name Portfile -exec egrep -H -m1 ansible {} \+
./python/py-ansible-compat/Portfile:name                py-ansible-compat
./python/py-ansible-core/Portfile:name                py-ansible-core
./python/py-ansible/Portfile:name                py-ansible
./python/ansible_select/Portfile:name                ansible_select
./sysutils/ansible-lint/Portfile:github.setup        ansible-community ansible-lint 6.9.1 v
./sysutils/ansible/Portfile:name            ansible
```

I checked `ansible-compat` and `ansible-lint`: they do not depend on removed subports of ansible

File: python/py-ansible/Portfile (hunk 2 above)
Comment (tsabirgaliev) [reply python/py-ansible/Portfile:33]: yep

**Comment by adfernandes** [comment]
LGTM

**Review by lpsinger — APPROVED** [review]

**Review by reneeotten — APPROVED** [review]

**Pull request merged**